src/loops/nested_loops/hard/q7/MainTest.java

```diff
@@ -0,0 +1,87 @@
+package loops.nested_loops.hard.q7;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Java
+    public Clause[] testSentence(){
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "5";
+        return new Clause[]{
+                new StringLiteral("Enter number of lines:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain(){BoxMeNow.main(new String[0]);}
+
+    static Stream<Integer> inputProvider(){
+        return Stream.of(1, 0, 5, 23, 2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void constructPatternCorrectly(int n) throws InvalidClauseException {
+        runWithInput(String.valueOf(n), clauseBuilder(n));
```

Comment:
This is missing an error message

Reply:
I figure the standard wrong format would work here since the program is all format.

Reply:
Same as the previous pull request, I think it is basically always better to specify an error message when possible.

Reply:
It just allows us to target the feedback towards the things we test for, as opposed to if they just spell a word or something wrong.